disorder-fsm/src/Disorder/FSM.hs

```diff
@@ -1,50 +1,87 @@
 {-# LANGUAGE NoImplicitPrelude #-}
 {-# LANGUAGE LambdaCase #-}
+{-# LANGUAGE GeneralizedNewtypeDeriving #-}
+{-# LANGUAGE ScopedTypeVariables #-}
 module Disorder.FSM (
+    module X
   -- * Transition type and constructor
-    Transition
+  , Transition
   , mkTransition
+  , ask
+  , get
+  , put
+  , modify
   -- * Combinators to make 'Transition' less trivial
   , goif
   , goto
   -- * Stubs for combinators arguments
   , always
   , sameState
+  -- * Lift functions for 'Gen'
+  , liftGen
+  -- * Assert helper functions
+  , assert
+  , assertMsg
+  , (===)
+  , (=/=)
   -- * Transition evaluation
   , runFSM
+  , runFSMCont
   ) where
 
 import           Prelude (Show(..))
 import           Data.Bool
-import           Data.String
-import           Data.Maybe
+import           Data.Eq
 import           Data.Function
 import           Data.Monoid
+import           Data.String
+import           Control.Exception.Base (AssertionFailed(..))
 import           Control.Monad
 import           Control.Monad.Catch
+import           Control.Monad.Cont
+import           Control.Monad.Trans.Cont
+import           Control.Monad.RWS.Strict hiding (state)
+
+import           Test.QuickCheck as X hiding ((===))
+import           Test.QuickCheck.Monadic as X hiding (assert)
 
 import           Test.QuickCheck.Gen
-import           Test.QuickCheck.Monadic
-import           Test.QuickCheck.Property
 
 
 -- | Defines a transition from state to state
 data Transition e s m = MkTransition {
     -- | Display name (used in case of failure)
     name :: String
```

Comment:
@etorreborre `name` is only used here in `Show` and `fail` both of which require `String`. Our `Text` would always be converted to `String` anyway.